### PR TITLE
Make SiPixelDigiToRaw a global module

### PR DESCRIPTION
**Pull request by Dr15Jones**
Also removed filling of timing and digi counts histograms, as those can not be done in a thread efficent manner.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: EventFilter/SiPixelRawToDigi/plugins/SiPixelDigiToRaw.cc [hunk 1]
```diff
@@ -16,136 +30,142 @@
...
 
 // -----------------------------------------------------------------------------
-SiPixelDigiToRaw::~SiPixelDigiToRaw() {
-  delete frameReverter_;
+std::shared_ptr<pr::Cache> 
+SiPixelDigiToRaw::globalBeginLuminosityBlock(edm::LuminosityBlock const&, 
+                                             edm::EventSetup const& es) const {
+  while(lock_.test_and_set(std::memory_order_acquire)); //spin
+  auto rel = [](std::atomic_flag* f) { f->clear(std::memory_order_release); };
+  std::unique_ptr<std::atomic_flag, decltype(rel)> guard(&lock_, rel);
```
Comment (slava77) [inline EventFilter/SiPixelRawToDigi/plugins/SiPixelDigiToRaw.cc]: I wish there were a simpler version compared to these 3 lines.

Comment (Dr15Jones) [reply EventFilter/SiPixelRawToDigi/plugins/SiPixelDigiToRaw.cc]: If you prefer, we could switch to a boost spinlock
https://www.boost.org/doc/libs/1_54_0/doc/html/atomic/usage_examples.html#boost_atomic.usage_examples.example_spinlock.usage

but that would add a dependency on boost.

**Comment by slava77** [comment]
@tsusa @dkotlins @veszpv 
please note that this PR removes some statistics accumulation functionality from SiPixelDigiToRaw:
- simple accumulation of input digis
- CPU timing histogramming

Both were not a part of the default configuration.
Please confirm that the removal is OK.

**Comment by slava77** [comment]
@Dr15Jones 
please update the PR description to highlight changes in the functionality of the code (other than a switch to global module, some debug statistics functionality is removed)

**Comment by dkotlins** [comment]
I think this is OK.
If I remember right the simple accumulation has not been used for a while.
Also the CPU histograms we have not been looking at them recently.
There are probably other tools which can measure the CPU usage better now.
Danek

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/24057	e1cd932
- mostly technical update to move SiPixelDigiToRaw to be a global module; some debug statistics functionality was removed, which appears to be OK
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by fabiocos** [comment]
+1

**Pull request merged**